### PR TITLE
wrap actions/attest-build-provenance

### PR DESCRIPTION
**Pull request by bdehamer**
Replace local implementation with a call to the public `actions/attest-build-provenance` action.

This will ensure that anyone who is still referencing the private beta version of the action is getting all of the updates and bug fixes of the public action.

Also includes a deprecation warning in the job summary:
image

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -37,7 +37,27 @@ outputs:
...
+        subject-digest: ${{ inputs.subject-digest }}
+        subject-name: ${{ inputs.subject-name }}
+        push-to-registry: ${{ inputs.push-to-registry }}
+        github-token: ${{ inputs.github-token }}
+    - id: translate-output
+      if: runner.os != 'Windows'
+      shell: bash
+      run: |
+        read -r line < ${{ steps.attest.outputs.bundle-path }}
+        echo "bundle=$line" >> $GITHUB_OUTPUT
```
Comment (bdehamer) [inline action.yml:58]: The primary difference between the early-access action and the public action is the way the action output is handled:

*  In the early-access action, the JSON-serialized bundle is passed back in the `bundle` output variable.
* In the new action, any attestation bundles are written to a file and the path to that file is passed back in the `bundle-path` output variable.

This bit translates the new scheme to the old one for backward compatibility. We don't handle Windows cause there's not easy way to make this work on both Windows and non-Windows runners.

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -30,6 +30,10 @@ jobs:
       - name: Checkout
         id: checkout
         uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4.1.4
+      - name: Update GH CLI
+        uses: dev-hanz-ops/install-gh-cli-action@v0.2.0
```
Comment (steiza) [inline .github/workflows/e2e.yml:34]: Minor: should we pin to a specific hash? Especially as this it the private beta / interim Action.

File: action.yml (hunk 1 above)
Comment (steiza) [reply action.yml:58]: The only user I'd worry about for the Windows platform is Homebrew... but I believe yesterday they switched to `actions/attest-build-provenance` so I think we can proceed!

**Review by steiza — APPROVED** [review]
I did not review the 101,076 lines you deleted 😐

File: action.yml (hunk 1 above)
Comment (bdehamer) [reply action.yml:58]: Also, I checked and they are *not* using the action output

**Pull request merged**